src/coreclr/vm/reflectioninvocation.cpp

```diff
@@ -1024,7 +805,7 @@ FCIMPL5(Object*, RuntimeMethodHandle::InvokeMethod,
             argDest = ArgDestination(pStackCopy, 0, NULL);
         }
 
-        InvokeUtil::CopyArg(th, &objs->GetAt(i), &argDest);
+        InvokeUtil::CopyArg(th, (PVOID **)&(args[i]), &argDest);
```

Comment:
Yes the original code needed that to modify\replace a `TypedReference` in the main `switch` statement, so it needed the element by reference. However, since then I removed the `TypedReference` `case` statement since it can't be boxed and I had no good way to test it. When we add the byref caller that doesn't box then I'll need to add back support for `TypedReference` which at that point will likely need to leverage the extra indirection.

> This can be args + i and drop the &

Sure - my slight preference for readability is `[i]` over `+1` but can change it. Perhaps not reset CI however for just that.

Reply:
I removed the extra level and changed the `CopyArg` signature to just take PVOID which is true for value types. When an object reference is needed then do the extra cast from PVOID to PVOID*. This I think makes it more clear.